packages/frontend/core/src/modules/peek-view/entities/peek-view.ts

```diff
@@ -110,11 +151,20 @@ export class PeekViewEntity extends Entity {
     target: ActivePeekView['target'],
     template?: TemplateResult
   ) => {
-    const resolvedInfo = resolvePeekInfoFromPeekTarget(target);
-    if (!resolvedInfo && !template) {
+    const resolvedInfo = resolvePeekInfoFromPeekTarget(target, template);
+    if (!resolvedInfo) {
       return;
     }
-    this._active$.next({ target, info: resolvedInfo, template });
+
+    const active = this._active$.value;
+
+    // if there is an active peek view and it is a doc peek view, we will navigate it first
+    if (active?.info.type === 'doc' && this.show$.value) {
```

Comment:
the effect must look weird I think, lol

Reply:
weird indeed, but confirmed with the product